openai/src/commonMain/kotlin/com/xebia/functional/xef/auto/llm/openai/OpenAI.kt

```diff
@@ -93,6 +93,7 @@ class OpenAI(internal val token: String) : AutoCloseable, AutoClose by autoClose
   }
 }
 
-fun String.toOpenAIModel(): OpenAIModel? {
-  return OpenAI.DEFAULT.supportedModels().find { it.name == this }
+fun String.toOpenAIModel(token: String): OpenAIModel {
+  val openAI = OpenAI(token)
+  return openAI.supportedModels().find { it.name == this } ?: openAI.GPT_3_5_TURBO_16K
```

Comment:
```suggestion
  return openAI.supportedModels().find { it.name == this } ?: openAI.DEFAULT_CHAT
```

Reply:
I think that to be able to use the `DEFAULT_CHAT` we need to modify the companion object inside the `OpenAI` class. Those default values use the `openAITokenFromEnv()` method.